### PR TITLE
Add tsd

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
- Added [tsd](https://github.com/SamVerschueren/tsd) to check generated type definitions
- Type fixes

https://oktainc.atlassian.net/browse/OKTA-370001
https://oktainc.atlassian.net/browse/OKTA-356468

Resolves: https://github.com/okta/okta-auth-js/issues/576

### CODE REVIEW

**Comment by shuowu** [comment]
You can add a `types` folder under `test`, then move the validation scripts there. The scripts folder is for build scripts.

**Comment by oleksandrpravosudko-okta** [comment]
LGTM - looking for 👍 from @shuowu.

File: scripts/tsd.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -xe
+
+cp ./test/types/* ./build/
```
Comment (shuowu) [inline scripts/tsd.sh:3]: We don't need to include tests (types) in the build.

File: test/types/auth.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  popupTitle: 'Hello Okta',
+};
+
+const authorizeOptions2: TokenParams = {
+  responseType: 'token',
+  idpScope: ['email', 'profile'],
+  maxAge: '100',
+};
+
+(async () => {
```
Comment (shuowu) [inline test/types/auth.test-d.ts:51]: So we need to wrap the test body in IIFE to test async APIs?

File: .eslintignore [hunk 3]
```diff
@@ -7,3 +7,5 @@ node_modules
 /build/cjs
 /samples/templates
 /samples/generated/webpack-spa/public/*-bundle.*
+/scripts
```
Comment (shuowu) [inline .eslintignore:10]: Can we remove these two lines from .eslintignore file?

File: package.json [hunk 4]
```diff
@@ -23,8 +23,9 @@
     "clean:build": "rimraf build && rimraf build2",
     "dev": "webpack --config webpack.dev.config.js",
     "lint": "eslint .",
+    "tsd": "scripts/tsd.sh",
```
Comment (shuowu) [inline package.json:26]: Can we config tsd to point it to the test/types folder? So we don't need the script anymore.

File: scripts/tsd.sh (hunk 1 above)
Comment (denysoblohin-okta) [reply scripts/tsd.sh]: Right.  Done

File: test/types/auth.test-d.ts (hunk 2 above)
Comment (denysoblohin-okta) [reply test/types/auth.test-d.ts:51]: Without wrapping linter returns error: `Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext' or 'system', and the 'target' option is set to 'es2017' or higher.ts(1378)`

**Review by shuowu — APPROVED** [review]

**Pull request closed without merging**